packages/tools/src/tools/annotation/planarFreehandROITool/renderMethods.ts

```diff
@@ -0,0 +1,300 @@
+
+const { pointsAreWithinCloseContourProximity } = polyline;
+
+/**
+ * Renders a `PlanarFreehandROIAnnotation` that is not currently being drawn or edited.
+ */
+function renderContour(
+  enabledElement: Types.IEnabledElement,
+  svgDrawingHelper: any,
+  annotation: PlanarFreehandROIAnnotation
+): void {
+  if (annotation.data.isOpenContour) {
+    this.renderOpenContour(enabledElement, svgDrawingHelper, annotation);
+  } else {
+    this.renderClosedContour(enabledElement, svgDrawingHelper, annotation);
+  }
+}
+
+/**
+ * Renders an closed `PlanarFreehandROIAnnotation` annotation.
+ */
+function renderClosedContour(
+  enabledElement: Types.IEnabledElement,
+  svgDrawingHelper: any,
+  annotation: PlanarFreehandROIAnnotation
+): void {
+  const { viewport } = enabledElement;
+
+  const settings = Settings.getObjectSettings(annotation, this.getToolName());
+
+  // Todo -> Its unfortunate that we have to do this for each annotation,
+  // Even if its unchanged. Perhaps we should cache canvas points per element
+  // on the tool? That feels very weird also as we'd need to manage it/clean
+  // them up.
+  const canvasPoints = annotation.data.polyline.map((worldPos) =>
+    viewport.worldToCanvas(worldPos)
+  );
+
+  const lineWidth = this.getStyle(settings, 'lineWidth', annotation);
+  // const lineDash = this.getStyle(settings, 'lineDash', annotation);
+  const color = this.getStyle(settings, 'color', annotation);
+
+  const options = {
+    color: color === undefined ? undefined : <string>color,
+    width: lineWidth === undefined ? undefined : <number>lineWidth,
+    connectLastToFirst: true,
+  };
+
+  const polylineUID = '1';
+
+  drawPolylineSvg(
+    svgDrawingHelper,
+    this.getToolName(),
+    annotation.annotationUID,
+    polylineUID,
+    canvasPoints,
+    options
+  );
+}
+
+/**
+ * Renders an open `PlanarFreehandROIAnnotation` annotation.
+ */
+function renderOpenContour(
+  enabledElement: Types.IEnabledElement,
+  svgDrawingHelper: any,
+  annotation: PlanarFreehandROIAnnotation
+): void {
+  const { viewport } = enabledElement;
+  const settings = Settings.getObjectSettings(annotation, this.getToolName());
+
+  // Its unfortunate that we have to do this for each annotation,
+  // Even if its unchanged. In the future we could cache the canvas points per
+  // element on the tool? That feels very weird also as we'd need to manage
+  // it/clean them up. Its a pre-optimisation for now and we can tackle it if it
+  // becomes a problem.
+  const canvasPoints = annotation.data.polyline.map((worldPos) =>
+    viewport.worldToCanvas(worldPos)
+  );
+
+  const lineWidth = this.getStyle(settings, 'lineWidth', annotation);
+  const color = this.getStyle(settings, 'color', annotation);
+
+  const options = {
+    color: color === undefined ? undefined : <string>color,
+    width: lineWidth === undefined ? undefined : <number>lineWidth,
+    connectLastToFirst: false,
+  };
+
+  const polylineUID = '1';
+
+  drawPolylineSvg(
```

Comment:
The logic for drawPolyLineSVG is almost the same for open/closed contour, can you extract them both in a common logic which accepts `connectLastToFirst`?

Reply:
I just tried doing this and its awkward as the methods at the end of renderOpenContour actually requires all the logic you could refactor out (also needs the canvasPoints, options, styles, etc). I could do it but it would require recalcation of world points etc.

To all your points, I've refactored out the settings + options bit, which is common, combining these two is not so useful though.